WinRT.Runtime/Interop/IAgileReference.netstandard2.0.cs

```diff
@@ -1,213 +1,213 @@
-        protected readonly ObjectReference<Vftbl> _obj;
-        public IntPtr ThisPtr => _obj.ThisPtr;
-        public ObjectReference<I> AsInterface<I>() => _obj.As<I>();
-        public A As<A>() => _obj.AsType<A>();
-
-        public IGlobalInterfaceTable(IObjectReference obj) : this(obj.As<Vftbl>()) { }
-        public IGlobalInterfaceTable(ObjectReference<Vftbl> obj)
-        {
-            _obj = obj;
-        }
-
-        public IntPtr RegisterInterfaceInGlobal(IObjectReference objRef, Guid riid)
-        {
-            ExceptionHelpers.ThrowExceptionForHR(_obj.Vftbl.RegisterInterfaceInGlobal(ThisPtr, objRef.ThisPtr, ref riid, out IntPtr cookie));
-            return cookie;
-
-        }
-
-        public void RevokeInterfaceFromGlobal(IntPtr cookie)
-        {
-            ExceptionHelpers.ThrowExceptionForHR(_obj.Vftbl.RevokeInterfaceFromGlobal(ThisPtr, cookie));
-        }
-
-        public IObjectReference GetInterfaceFromGlobal(IntPtr cookie, Guid riid)
-        {
-            ExceptionHelpers.ThrowExceptionForHR(_obj.Vftbl.GetInterfaceFromGlobal(ThisPtr, cookie, ref riid, out IntPtr ptr));
-            try
-            {
-                return ComWrappersSupport.GetObjectReferenceForInterface(ptr);
-            }
-            finally
-            {
-                MarshalInspectable.DisposeAbi(ptr);
-            }
-        }
-    }
-}
+﻿using System;
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+using System.Text;
+
+namespace WinRT.Interop
+{
+    [WindowsRuntimeType]
+    [Guid("C03F6A43-65A4-9818-987E-E0B810D2A6F2")]
+    internal interface IAgileReference
+    {
+        IObjectReference Resolve(Guid riid);
+    }
+
+    [WindowsRuntimeType]
+    [Guid("94ea2b94-e9cc-49e0-c0ff-ee64ca8f5b90")]
+    public interface IAgileObject
+    {
+    }
+
+    [WindowsRuntimeType]
+    [Guid("00000146-0000-0000-C000-000000000046")]
+    internal interface IGlobalInterfaceTable
+    {
+        IntPtr RegisterInterfaceInGlobal(IObjectReference objRef, Guid riid);
+        void RevokeInterfaceFromGlobal(IntPtr cookie);
+        IObjectReference GetInterfaceFromGlobal(IntPtr cookie, Guid riid);
+    }
+}
+
+namespace ABI.WinRT.Interop
+{
+    using global::WinRT;
+    using WinRT.Interop;
+
+    [Guid("C03F6A43-65A4-9818-987E-E0B810D2A6F2")]
+    internal unsafe class IAgileReference : global::WinRT.Interop.IAgileReference
+    {
+        [Guid("C03F6A43-65A4-9818-987E-E0B810D2A6F2")]
+        public struct Vftbl
+        {
+            public global::WinRT.Interop.IUnknownVftbl IUnknownVftbl;
+            private void* _Resolve;
+            public delegate* stdcall<IntPtr, ref Guid, out IntPtr, int> Resolve { get => (delegate* stdcall<IntPtr, ref Guid, out IntPtr, int>)_Resolve; set => _Resolve = value; }
+
+            public static readonly Vftbl AbiToProjectionVftable;
+            public static readonly IntPtr AbiToProjectionVftablePtr;
+
+#if NETSTANDARD2_0
```

Comment:
cruft: everywhere we've forked sources, we can get rid of conditional compilation